### PR TITLE
Add examples in error explanation E0267 and E0268

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Manishearth

### CODE REVIEW

File: src/librustc/diagnostics.rs [hunk 1]
```diff
@@ -602,15 +602,47 @@ const Y: u32 = X;
...
-be taken.
+be taken. Erroneous code example:
+
+```
+fn some_func() {
+    break; // error: `break` outside of loop
+}
+```
+
+So please check you use `break` and `continue` only in loops. Example:
```
Comment (Manishearth) [inline src/librustc/diagnostics.rs]: Please verify that you are using ... only in loops

**Comment by GuillaumeGomez** [comment]
Updated.

**Comment by Manishearth** [comment]
r=me post squash

**Comment by GuillaumeGomez** [comment]
Squashed.

**Comment by GuillaumeGomez** [comment]
@Manishearth: Retry ?

**Pull request merged**